rules/additional_scheme_info.bzl

```diff
@@ -0,0 +1,56 @@
+""" TODO """
+
+load("@build_bazel_rules_apple//apple:providers.bzl", "AppleBundleInfo")
+
+AdditionalSchemeInfo = provider(
+    """   
+    TODO
+    """,
+    fields = {
+        "additional_scheme_info": "TODO",
+    },
+)
+
+def _additional_scheme_info_impl(ctx):
+    test_action_targets = [
+        struct(
+            name = test_target[AppleBundleInfo].bundle_name,
+            environment_variables = ctx.attr.test_environment,
+        )
+        for test_target in ctx.attr.test_bundle_targets
+    ]
+
+    xcode_scheme = struct(
+        scheme_build_action_target = ctx.attr.build_target[AppleBundleInfo].bundle_name,
```

Comment:
nit: I think it might be a bit simpler for these to be fields on the `AdditionalSchemeInfo` provider itself.

Reply:
Yup, yup, totally.

And done.